internal/controllers/manager.go

```diff
@@ -85,14 +93,17 @@ func (m *TiltServerControllerManager) SetUp(ctx context.Context, st store.RStore
 		LeaderElection:   false,
 		LeaderElectionID: "tilt-apiserver-ctrl",
 
-		ClientBuilder:           m.builder,
+		ClientDisableCacheFor:   m.uncachedObjects,
 		Logger:                  logr,
 		GracefulShutdownTimeout: &timeout,
 	})
 	if err != nil {
 		return fmt.Errorf("unable to create controller manager: %v", err)
 	}
 
+	// provide the deferred client with the real client now that it has been initialized
+	m.deferredClient.initialize(mgr.GetClient())
```

Comment:
`ClientBuilder` is gone and `ctrl.NewManager` can handle uncached objects, so this just explicitly initializes the deferred client after setting that up. Uncached objects are injected via `ProvideUncachedObjects` (currently nil for release, except in tests where we explicitly call this constructor and pass Filewatch)